src/http_server.cc

```diff
@@ -67,6 +67,44 @@ namespace triton { namespace server {
     }                                                 \
   } while (false)
 
+#define HTTP_RESPOND_WITH_CODE(REQ, CODE, MSG)    \
+  do {                                            \
+    EVBufferAddErrorJson((REQ)->buffer_out, MSG); \
+    evhtp_send_reply((REQ), CODE);                \
+    return;                                       \
+  } while (false)
+
+namespace {
+void
+EVBufferAddErrorJsonInternal(evbuffer* buffer, const char* message)
+{
+  triton::common::TritonJson::Value response(
+      triton::common::TritonJson::ValueType::OBJECT);
+  response.AddStringRef("error", message, strlen(message));
+
+  triton::common::TritonJson::WriteBuffer buffer_json;
+  response.Write(&buffer_json);
+
+  evbuffer_add(buffer, buffer_json.Base(), buffer_json.Size());
+}
+
+void
+EVBufferAddErrorJson(evbuffer* buffer, const std::string& error_str)
```

Comment:
currently you don't need the overload for string here vs the const char *message - would this be for the future?

currently you could get away with EvBufferAddError(const char *message), and EvBufferAddError(TritonServer_Error*err)